File: 01 classes & objects/classes_and_objects.java
public class classes_and_objects {
    
    public static class User{
        String username;
        String Password;
    
        public String name;
        public int age;
        public boolean is_admin;
    
        String getUsername(){
            return username;
        }
        String getPassword(){
            return Password;
        }
        void setUsername(String username){
            this.username = username;
        }
        void setPassword(String password){
            this.Password = password;
        }
    
        public User(String name, int age, boolean is_admin){
            this.name = name;
            this.age = age;
            this.is_admin = is_admin;
        }
    
        void display_info(){
            System.out.println(name + " " + age + " " + is_admin);
        }
    
        void changePassword(String newPassword){
            Password = newPassword;
        }
        void increaseAge(){
            age += 1;
        }
    }


    public static void main(String[] args) {
        User Aditya = new User("aditya", 20, true);
        Aditya.display_info();
        Aditya.setUsername("adityapourush");
        Aditya.changePassword("<PASSWORD>");
        System.out.println(Aditya.getUsername());
        System.out.println(Aditya.getPassword());
        Aditya.increaseAge();
        Aditya.display_info();
    }
}

File: 01 classes & objects/c++/classes and objects.cpp
#include <iostream>
#include <string>

using namespace std;

class User {
    private:
    string username;
    string password;

    public:
    string name;
    int age;
    string role;

    // getters and setters for private attributes
    string getUsername() {
        return username;
    }
    string getPassword() {
        return password;
    }
    void setUsername(string username) {
        this->username = username;
    }
    void setPassword(string password) {
        this->password = password;
    }

    // Constructor for public attributes
    User(string name, int age, string role) {
        cout << "Making an object from class User" << endl;
        this->name = name;
        this->age = age;
        this->role = role;
    }

    // member function
    void displayInfo() {
        cout << "User Information:" << endl;
        cout << "Name: " << name << endl;
        cout << "Age: " << age << endl;
        cout << "Role: " << role << endl;
    }

    // member function
    void changePassword(string newPassword) {
        password = newPassword;
    }
    void increaseAge() {
        age++;
    }
};

class SmartDevice {
    private:
        string manufacturer;
        string model;
    public:
        string name;
        bool isOn;
        bool isConnected;
        int batteryLevel;
    
    SmartDevice(string name, string manufacturer, string model) {
        this->name = name;
        this->manufacturer = manufacturer;
        this->model = model;
        this->isOn = false;
        this->isConnected = false;
        this->batteryLevel = 100;
    }
    void turnOn() {
        if(!isOn){
            isOn = true;
        }
        cout << name << " is now turned on." << endl;
    }
    void turnOff() {
        if(isOn){
            isOn = false;
        }
        cout << name << " is now turned off." << endl;
    }

    void connectToNetwork() {
        if(!isConnected){
            isConnected = true;
        }
        cout << name << " is now connected to the network." << endl;
    }
    void disconnectFromNetwork() {
        if(isConnected){
            isConnected = false;
        }
        cout << name << " is now disconnected from the network. " << endl;
    }

    void displayInfo() {
        cout << "Device Information:" << endl;
        cout << "Name: " << name << endl;
        cout << "Manufacturer: " << manufacturer << endl;
        cout << "Model: " << model << endl;
        cout << "Status: " << (isOn ? "On" : "Off") << endl;
        cout << "Battery Level: " << batteryLevel << "%" << endl;
        cout << "Connected to Network: " << (isConnected ? "Yes" : "No") << endl;
    }
};

int main() {
    User Aditya("Aditya Raj Pourush", 20, "Admin");
    Aditya.displayInfo();
    Aditya.increaseAge();
    Aditya.displayInfo();
    cout << "Username: " << Aditya.getUsername() << endl;
    Aditya.setUsername("adityapourush");
    Aditya.setPassword("asdf1234");
    cout << "Username: " << Aditya.getUsername() << endl;
    cout << "Password: " << Aditya.getPassword() << endl;

    SmartDevice Television("Samsung TV", "Samsung", "QN90B");
    Television.turnOn();
    Television.connectToNetwork();
    Television.displayInfo();
    Television.turnOff();
    Television.displayInfo();
    return 0;
}